src/Ui.Playground/Internal/PlaygroundPageModel.cs

```diff
@@ -60,19 +62,30 @@ public string Render()
         return _playgroundCSHtml;
     }
 
-    // https://html.spec.whatwg.org/multipage/scripting.html#restrictions-for-contents-of-script-elements
-    private static string StringEncode(string value) => value
-        .Replace("\\", "\\\\")  // encode  \  as  \\
-        .Replace("<", "\\x3C")  // encode  <  as  \x3C   -- so "<!--", "<script" and "</script" are handled correctly
-        .Replace("'", "\\'")    // encode  '  as  \'
-        .Replace("\"", "\\\""); // encode  "  as  \"
-
-    private static string JsonSerialize(object value)
+    private string JsonSerialize(object value)
     {
+        if (_services.GetService(typeof(IGraphQLSerializer)) is IGraphQLSerializer serializer)
+        {
+            using var stream = new MemoryStream();
+            serializer.WriteAsync(stream, value).Wait();
+
+            stream.Position = 0;
+
+            using var reader = new StreamReader(stream, Encoding.UTF8);
+            return reader.ReadToEnd();
+        }
```

Comment:
FYI, GraphQL.NET provides a synchronous method as well:

```suggestion
        var serializer = _services.GetService<IGraphQLTextSerializer>();
        if (serializer != null)
        {
            return serializer.Serialize(value);
        }
```